### PR TITLE
Subsystems tests - domains, ssl, sasl + fix

### PR DESCRIPTION
**Pull request by honza889**
Tests of Domain, Sasl and SSL.
* Sasl and SSL use only server side from Elytron, client side not available yet. (?)
* Digest Sasl Authentication ignored yet - waiting for AvailableRealmsCallback support in Digest or PropertiesSaslServerFactory delegating filter in subsystem.

Depends on https://github.com/wildfly-security/wildfly-elytron/pull/405

### CODE REVIEW

File: src/test/java/org/wildfly/extension/elytron/DomainTestCase.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+    private KernelServices services = null;
+
+    private String filesystemFile = RealmsTestCase.class.getResource("/org/wildfly/extension/elytron/filesystem-realm").getFile();
+    private String propertiesFile = RealmsTestCase.class.getResource("/org/wildfly/extension/elytron/testingrealm1-users.properties").getFile();
+
+    private void init() throws Exception {
+        new ClassLoadingAttributeDefinitionsMock(); // mock classloader obtaining
+        String subsystemXml = "<subsystem xmlns=\"" + ElytronExtension.NAMESPACE + "\">\n" +
+                "    <security-domains>\n" +
```
Comment (darranl) [inline src/test/java/org/wildfly/extension/elytron/DomainTestCase.java]: I think I would prefer to keep the XML in an XML file that is read in by the test.  That way when working with IDEs we edit source code as source code and XML within an XML editor.

File: src/test/java/org/wildfly/extension/elytron/SslTestCase.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+    private final int TESTING_PORT = 18201;
+
+    public SslTestCase() {
+        super(ElytronExtension.SUBSYSTEM_NAME, new ElytronExtension());
+    }
+
+    private KernelServices services = null;
+
+    private String keyStoreFile = SslTestCase.class.getResource("/org/wildfly/extension/elytron/testingCaJks.keystore").getFile();
```
Comment (darranl) [inline src/test/java/org/wildfly/extension/elytron/SslTestCase.java]: Is using the fully qualified name really necessary? By using the SslTestCase class to obtain the resource it should be possible to obtain the resource relative to that classes package already.

File: src/test/java/org/wildfly/extension/elytron/SslTestCase.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+            Assert.fail(services.getBootError().toString());
+        }
+
+        ServiceName serverServiceName = Capabilities.SSL_CONTEXT_RUNTIME_CAPABILITY.getCapabilityServiceName("ServerSslContext");
+        SSLContext serverSslContext = (SSLContext) services.getContainer().getService(serverServiceName).getValue();
+        Assert.assertNotNull(serverSslContext);
+        SSLServerSocketFactory serverSocketFactory = serverSslContext.getServerSocketFactory();
+
+        /*
+        ServiceName clientServiceName = Capabilities.SSL_CONTEXT_RUNTIME_CAPABILITY.getCapabilityServiceName("ClientSslContext");
```
Comment (darranl) [inline src/test/java/org/wildfly/extension/elytron/SslTestCase.java]: If this is not required in the first implementation of the test is there a reason to keep it?

If there is I would suggest adding a comment so in the future we know why it is commented out, if not probably better to just remove it now.

**Comment by darranl** [comment]
Added a couple of comments but also: -

So far subsystem testing is starting to add some more key and trust store files, I think we need to make sure that we add a README that describes the contents of these files and how they are created / structure etc so in the future we can understand them if tests involving them subsequent stop working.  Also if at all possible we have a CA set up in the main WildFly Elytron project - it would be better if the smaller projects can borrow files from Elytron so we have one central place where we really manage the SSL related files.

**Comment by darranl** [comment]
Should have also said - apart from that - looks good ;-)

**Comment by honza889** [comment]
Fixed by notes. I have replaced my testing keystores by keystores from Elytron CA, so they should not be problem anymore.

**Comment by honza889** [comment]
+ unified TestEnvironment - standalone TestEnvironment in IdentityOperationTestCase replaced by common one

**Pull request merged**